parsl/executors/status_handling.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import threading
+from abc import abstractmethod
+from typing import List, Any, Dict
+
+from parsl.providers.provider_base import JobStatus, ExecutionProvider, JobState
+
+
+logger = logging.getLogger(__name__)
+
+
+class StatusHandlingMixin(object):
+    tasks = None
+    provider = None
+
+    def __init__(self, provider=None):
```

Comment:
This __init__ signature makes me uncomfortable: classes need to call super().__init__(provider) and assumes that the MRO makes super().__init__() be the StatusHandlingMixin, which feels fragile to me in the presence of subclassing.

Maybe it's more reasonable to require that `self.provider`  exists, so that this code does not store a second copy in `self._provider`.

Reply:
I think the reasonable solution is to have StatusHandling and NoStatusHandling be abstract classes rather than mixins. I don't see a good reason any more why they should be mixins, since they are very specific and unlikely to be used anywhere outside of an implementation of a ParslExecutor. I'll update the PR.

Reply:
That said, it's probably wise in general if mixins were conventionally restricted to only have trivial constructors.